### PR TITLE
`CALSLR_ROEBBER`: Instead of calling BOUND incorrectly, use min and max

### LINKED ISSUE #672: CALSLR_ROEBBER in UPP_PHYSICS.f steps on memory by calling BOUND incorrectly
`CALSLR_ROEBBER` in UPP_PHYSICS.f is sending an array with the wrong bounds to the BOUND function, causing it to step all over memory. This issue was uncovered in testing of https://github.com/ufs-community/ufs-weather-model/pull/1642

```f90
real,dimension(ista:iend,jsta:jend)    :: T2M,RH2M
...
CALL BOUND(RH2M,H1,H100)
```

The BOUND subroutine expects an IM,JM array:

```f90
SUBROUTINE BOUND(FLD,FMIN,FMAX)
REAL,intent(in) :: FMAX, FMIN
REAL,intent(inout) :: FLD(IM,JM)
```

### LINKED ISSUE #674: RHSFC `BOUND` call uses incorrect array bounds, causing out-of-bounds write
The RHSFC in SURFCE.f is allocated like this:

```f90
         allocate(zsfc(ista:iend,jsta:jend),  psfc(ista:iend,jsta:jend),  tsfc(ista:iend,jsta:jend)&
                 ,rhsfc(ista:iend,jsta:jend), thsfc(ista:iend,jsta:jend), qsfc(ista:iend,jsta:jend))
```

and it is sent to `BOUND`

```f90
            CALL BOUND(RHSFC,H1,H100)
```

which expects an im,jm array

```f90
      SUBROUTINE BOUND(FLD,FMIN,FMAX)
...
      REAL,intent(inout) :: FLD(IM,JM)
```

### PR DESCRIPTION
**Pull request by SamuelTrahanNOAA**
Fixes #672 by calling min and max in CALSLR_ROEBBER instead of calling BOUND incorrectly.

Fixes #674 the same way.

### CODE REVIEW

**Comment by SamuelTrahanNOAA** [comment]
I just noticed I added a blank line by accident. The ufs-weather-model regression tests are already running, so I'd rather not remove the accidental blank line. It can't affect anything since the Fortran parser eliminates blank lines before the compiler sees them.

**Comment by SamuelTrahanNOAA** [comment]
Apparently the author year is wrong too. That seems somewhat more important than a blank line.

**Comment by WenMeng-NOAA** [comment]
@SamuelTrahanNOAA Can you sync your branch with the latest UPP develop?

File: sorc/ncep_post.fd/SURFCE.f [hunk 1]
```diff
@@ -371,7 +370,7 @@ SUBROUTINE SURFCE
                jj = jsta+j-1
                do i=1,iend-ista+1
                ii = ista+i-1
-                 datapd(i,j,cfld) = RHSFC(ii,jj)
+                 datapd(i,j,cfld) = max(H1,min(H100,RHSFC(ii,jj)))
```
Comment (WenMeng-NOAA) [inline sorc/ncep_post.fd/SURFCE.f:375]: @SamuelTrahanNOAA Can you add a checking for undefined grids in calculation?
such as: if (RHSFC(ii,jj) /= spval)

Comment (WenMeng-NOAA) [reply sorc/ncep_post.fd/SURFCE.f:375]: @SamuelTrahanNOAA Please add a change log.

Comment (SamuelTrahanNOAA) [reply sorc/ncep_post.fd/SURFCE.f:375]: I have done so.

**Review by WenMeng-NOAA — APPROVED** [review]

**Comment by WenMeng-NOAA** [comment]
The UPP RT tests were completed on WCOSS2, Hera and Orion without changed results. This PR is ready for merging.

**Pull request merged**